src/utils/date.ts

```diff
@@ -0,0 +1,34 @@
+import { isRight } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+
+export const dateRegex = /^(?<year>[12]\d{3})-(?<month>0[1-9]|1[0-2])-(?<day>0[1-9]|[12]\d|3[01])$/;
+const dateRegexGroups = t.interface({
+  day: t.string,
+  month: t.string,
+  year: t.string
+});
+
+export const isAdult = (dateOfBirth: string): boolean => {
```

Comment:
I'd refactor this method to take currentDate as input. This will simplify tests of isAdult() that should be deterministic and not use `new Date()` (tests should use hardcoded dates)

something like --> `const isOlderThan = (years: number) => (dateOfBirth: Date, when: Date) => { ... }` 

moreover I'd move the date parsing into the caller and, if we already include something like date-fns (or momentjs) use a library rather than a regex to parse the input date

Reply:
Ok, I'll use `date-fns` that is already present into the project to handle date parsing

Reply:
Fixed in 34282a4